sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -692,4 +689,17 @@ public final class StorageBlobClient: PipelineClient {
         StorageBlobClient.manager.add(transfer: blobTransfer)
         return blobTransfer
     }
+
+    // MARK: Internal Methods
+
+    internal func addCancellationToken(to context: PipelineContext, from options: AzureOptions?) {
+        if let cancellationToken = options?.cancellationToken {
```

Comment:
Slight change in behavior here - I'm not sure which is better. Previously, if the customer provided a cancellation token without a timeout, we would apply the globalTimeout to the existing token. Now, the global timeout is only applied if no cancellation token is supplied at all.

I suspect the previous behavior is preferable - we probably should not make it easy to do network operations with an unlimited timeout. In the previous behavior, the customer could make use of the default timeout and still pass in a token if they wished to cancel manually.

Reply:
I'm also relocating this logic to be centralized in the `PipelineContext` object.